packages/@aws-cdk/aws-lambda-event-sources/lib/sqs.ts

```diff
@@ -0,0 +1,24 @@
+import lambda = require('@aws-cdk/aws-lambda');
+import sqs = require('@aws-cdk/aws-sqs');
+
+export interface SqsEventSourceProps {
+  batchSize?: number;
+}
+
+export class SqsEventSource implements lambda.IEventSource {
+  constructor(readonly queue: sqs.QueueRef, private readonly props: SqsEventSourceProps = { }) {
+    if (this.props.batchSize !== undefined && (this.props.batchSize < 1 || this.props.batchSize > 10)) {
+      throw new Error(`Maximum batch size must be between 1 and 10 inclusive (given ${this.props.batchSize})`);
+    }
+  }
+
+  public bind(target: lambda.FunctionRef) {
+    new lambda.EventSourceMapping(target, `SqsEventSource:${this.queue.uniqueId}`, {
```

Comment:
Since SNS and S3 have it, I would like to see the reciprocal function on the SQS queue as well.

Reply:
It might be tricky due to dependency cycle. Opened https://github.com/awslabs/aws-cdk/issues/1064 to track